wpilibc/athena/src/DriverStation.cpp

```diff
@@ -584,9 +582,13 @@ void DriverStation::Run() {
   while (m_isRunning) {
     HAL_WaitForDSData();
     GetData();
+    // notify IsNewControlData variables
+    m_newControlData = true;
+
+    // notify WaitForData block
     {
       std::lock_guard<priority_mutex> lock(m_waitForDataMutex);
-      m_updatedControlLoopData = true;
+      m_waitForDataPredicate = true;
     }
     m_waitForDataCond.notify_all();
```

Comment:
Heh, you're right: 

"The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock."